frontend/catalyst/api_extensions/quantum_operators.py

```diff
@@ -439,6 +521,64 @@ def wires(self):
         total_wires = sum((op.wires for op in self.regions[0].quantum_tape.operations), [])
         return total_wires
 
+    def decomposition(self):
+        """Resolve the Adjoint region by propagating the adjoint modifier to nested operations
+        in reverse."""
+        assert len(self.regions) == 1, "Expected a single nested region for HybridAdjoint"
+
+        # While catalyst.adjoint would be just as valid since it is PL compatible for single ops,
+        # going to PennyLane's adjoint skips unnecessarily re-queing the base operation in each
+        # HybridOp's nested tape when qjit is not active.
+        return [qml.adjoint(op) for op in reversed(self.regions[0].quantum_tape.operations)]
+
+
+class Adjoint:
+    """This class provides near identical behaviour as PennyLane for adjoint instances with only a
+    single base operation. Additionally, it provides the same functionality as HybridAdjoint."""
+
+    def __new__(cls, base_op, _tracing_artifacts):
+        if isinstance(base_op, Operation) and isinstance(base_op, Observable):
+            return object.__new__(AdjointOpObs)
+        if isinstance(base_op, Operation):
+            return object.__new__(AdjointOperation)
+        elif isinstance(base_op, Observable):
+            object.__new__(AdjointObs)
+
+        return object.__new__(AdjointBase)
+
+    def __init__(self, base_op, tracing_artifacts):
+        # Grab the constructor from the PennyLane base class.
+        super().__init__(base_op)
+        HybridAdjoint.__init__(self, *tracing_artifacts)
+
+    # These attributes are provided by the mixin class.
+    # pylint: disable=no-member
+    def _flatten(self):
+        tracing_artifacts = (self.in_classical_tracers, self.out_classical_tracers, self.regions)
+        return (self.base, tracing_artifacts), tuple()
+
+    @classmethod
+    def _unflatten(cls, data, _):
+        return cls(*data)
+
+
+# HybridAdjoint is also mixed in because the PL class needs to sit between Adjoint & HybridAdjoint.
+# Thus Adjoint cannot be made to inherit from HybridAdjoint directly.
+class AdjointOperation(Adjoint, pl_adjoint_module.AdjointOperation, HybridAdjoint):
+    """Replicate mixin class structure from PennyLane for Operations."""
+
+
+class AdjointObs(Adjoint, pl_adjoint_module.AdjointObs, HybridAdjoint):
```

Comment:
I may be missed it in `test_adjoint.py`; where is this adjoint(obs) tested?

Reply:
Good question, I don't know. I would hope it's part of the PL test suite that I copied over 👀

Reply:
It seems to be tested in a few places such as [here](https://github.com/PennyLaneAI/catalyst/blob/fc1b0320ed347fa9f69c68e506c7155f87db2899/frontend/test/pytest/test_adjoint.py#L1201). There are also tests for HermitianObs :+1: